### ISSUE TITLE
do not insert pairs when cursor is in the middle of a text

### ISSUE THREAD

**Issue by elianiva** [windwp/nvim-autopairs#17]
Hi there! I would love to be able to disable the insertion of matching pair when I'm in the middle of a word. For example:

```lua
require"some_text"|.foobar() -- | is the cursor, do not insert matching pair here

require"some_text" |.foobar() -- | is the cursor, do not insert matching pair here

require"some_text".foobar| -- | is the cursor, insert matching pair here

(|) -- | is the cursor, insert matching pair here, also applies to all matches like [] and {}
```

Basically, I don't want it to insert a matching pair if the right side of the cursor is not empty or it's not a symbol like what I've defined in the setup. (`(){}[]`). Thanks in advance!

edit: this seems to work already, but it still inserts a pair if the next character is a `.`

**Comment by windwp** [comment]
currently it will not add pairs if next character is a word. 

dot . is a operator like + - then it will add a pairs

**Comment by elianiva** [comment]
can I disable that behaviour?

**Comment by windwp** [comment]
you can make a PR for it :)

**Comment by elianiva** [comment]
ok, I'll make one later

**Comment by elianiva** [comment]
closed by #18
